examples/examples_test.go

```diff
@@ -0,0 +1,261 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"os"
+	"os/exec"
+
+	"github.com/tektoncd/pipeline/pkg/names"
+	knativetest "knative.dev/pkg/test"
+)
+
+const (
+	timeoutSeconds = 600
+	sleepBetween   = 10
+
+	// we may want to consider either not running examples that require registry access
+	// or doing something more sophisticated to inject the right registry in when folks
+	// are executing the examples
+	horribleHardCodedRegistry = "gcr.io/christiewilson-catfactory"
+)
+
+// formatLogger is a printf style function for logging in tests.
+type formatLogger func(template string, args ...interface{})
+
+// cmd will run the command c with args and if input is provided, that will be piped
+// into the process as input
+func cmd(logf formatLogger, c string, args []string, input string) (string, error) {
+	// Remove braces from args when logging so users can see the complete call
+	logf("Executing %s %v", c, strings.Trim(fmt.Sprint(args), "[]"))
+
+	cmd := exec.Command(c, args...)
+	cmd.Env = os.Environ()
+
+	if input != "" {
+		cmd.Stdin = strings.NewReader(input)
+	}
+
+	var stderr, stdout bytes.Buffer
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		logf("couldn't run command %s %v: %v, %s", c, args, err, stderr.String())
+		return "", err
+	}
+
+	return stdout.String(), nil
+}
+
+// eraseClusterTasks will erase all cluster tasks which do not get cleaned up by namespace.
+func eraseClusterTasks(logf formatLogger) {
+	if _, err := cmd(logf, "kubectl", []string{"delete", "--ignore-not-found=true", "clustertasks.tekton.dev", "--all"}, ""); err != nil {
+		logf("couldn't delete cluster tasks: %v", err)
+	}
+}
+
+// getYamls will look in the directory in examples indicated by version and run for yaml files
+func getYamls(t *testing.T, version, run string) []string {
+	t.Helper()
+	_, filename, _, _ := runtime.Caller(0)
+
+	// Don't read recursively; the only dir within these dirs is no-ci which doesn't
+	// want any tests run against it
+	files, err := ioutil.ReadDir(path.Join(path.Dir(filename), version, run))
+	if err != nil {
+		t.Fatalf("Couldn't read yaml files from %s/%s/%s: %v", path.Dir(filename), version, run, err)
+	}
+	yamls := []string{}
+	for _, f := range files {
+		if matches, _ := filepath.Match("*.yaml", f.Name()); matches {
+			yamls = append(yamls, f.Name())
+		}
+	}
+	return yamls
+}
+
+// replaceDockerRepo will look in the content f and replace the hard coded docker
+// repo with the on provided via the KO_DOCKER_REPO environment variable
+func replaceDockerRepo(t *testing.T, f string) string {
+	t.Helper()
+	r := os.Getenv("KO_DOCKER_REPO")
+	if r == "" {
+		t.Fatalf("KO_DOCKER_REPO must be set")
```

Comment:
Should we error out or `skip` ? (like we do on the e2e go tests — mainly to not break openshift-pipelines CI :stuck_out_tongue_closed_eyes: )